src/test/script_tests.cpp

```diff
@@ -1817,7 +1817,14 @@ BOOST_AUTO_TEST_CASE(bip341_keypath_test_vectors)
         }
 
     }
+}
 
+BOOST_AUTO_TEST_CASE(compute_tapbranch)
+{
+    uint256 hash1 = uint256S("8ad69ec7cf41c2a4001fd1f738bf1e505ce2277acdcaa63fe4765192497f47a7");
+    uint256 hash2 = uint256S("f224a923cd0021ab202ab139cc56802ddb92dcfc172b9212261a539df79a112a");
+    uint256 result = uint256S("a64c5b7b943315f9b805d7a7296bedfcfd08919270a1f7a1466e98f8693d8cd9");
+    BOOST_CHECK_EQUAL(ComputeTapbranchHash(hash1, hash2), result);
```

Comment:
I'd also appreciate a test case for `ComputeTapleafHash` since this PR touches the function.

Reply:
Add:

```c++
    BOOST_CHECK_EQUAL(ComputeTapbranchHash(hash2, hash1), result);

    const uint8_t script[6] = {'f','o','o','b','a','r'};
    uint256 tlc0 = uint256S("edbc10c272a1215dcdcc11d605b9027b5ad6ed97cd45521203f136767b5b9c06");
    uint256 tlc2 = uint256S("8b5c4f90ae6bf76e259dbef5d8a59df06359c391b59263741b25eca76451b27a");

    BOOST_CHECK_EQUAL(ComputeTapleafHash(0xc0, Span(script)), tlc0);
    BOOST_CHECK_EQUAL(ComputeTapleafHash(0xc2, Span(script)), tlc2);
```

?

Could be done in a later PR though.

Reply:
was a bit more interested that it didn't somehow change the definition(would be caught in other tests surely but still!)

```
BOOST_AUTO_TEST_CASE(compute_tapleaf)
{
    const std::vector<uint8_t> script = {'f','o','o','b','a','r'};
    uint256 tlc0 = uint256S("edbc10c272a1215dcdcc11d605b9027b5ad6ed97cd45521203f136767b5b9c06");
    uint256 tlc2 = uint256S("8b5c4f90ae6bf76e259dbef5d8a59df06359c391b59263741b25eca76451b27a");

    BOOST_CHECK_EQUAL(ComputeTapleafHash(0xc0, CScript(script.begin(), script.end())), tlc0);
    BOOST_CHECK_EQUAL(ComputeTapleafHash(0xc2, CScript(script.begin(), script.end())), tlc2);
}
```
passes on master. Your suggestion can be PR'd later.

Reply:
Done in #26934.